src/domain/domain.ts

```diff
@@ -23,8 +23,8 @@ export type DomainAggregate = Record<string, DomainAggregateItem | null>
 
 export type AddDomain = {
   name: string
-  target: AddDomainTarget
-  programType?: EntityType.Instance | EntityType.Program
+  target: AddDomainTarget | EntityType.Program | EntityType.Instance
+  programType: EntityType.Instance | EntityType.Program
```

Comment:
if `target` prop now accepts `ipfs` `program` and `instance`. I wouldn't require to pass a `programType` prop as it is redundant. If it is still needed as part of the aggregate, it can be autocalculated in the `add` method

Reply:
fixed